### ISSUE TITLE
`[environment] root` from an auto-discovered `ty.toml` is ignored for PEP 723 scripts (works only via `--config-file`/`TY_CONFIG_FILE`)

### ISSUE THREAD

**Issue by Andrew-Tierno** [astral-sh/ty#4324]
## Summary

In a repo whose only ty config is a bare, standalone `ty.toml` — no
`pyproject.toml` anywhere — the
documented escape hatch for the PEP-723 isolation default,
`[environment] root = ["."]`, **silently does nothing when the config file is
reached through plain auto-discovery**. The byte-identical `ty.toml` takes
effect only when passed explicitly via `--config-file` or `TY_CONFIG_FILE`.

The same split applies to the `[rules]` severity table: under auto-discovery a
PEP-723 script's `unresolved-import` lands at the hard-coded `error` and gets
no first-party search path; under explicit `--config-file` the configured
`warn` applies and (with `environment.root` set) the first-party search path
is restored. So a bare-`ty.toml` repo has no config-level fix at all — only
adding `--config-file` at every invocation site, which is easy to miss even
once (we had it un-set at three: pre-commit, an advisory CI step, and a
stricter CI gate).

## Reproduction

Three files, no `pyproject.toml`.

`ty.toml`:

```toml
[rules]
unresolved-import = "warn"
```

`plain.py` (control — ordinary module, no PEP 723 header):

```python
import sibling_package  # deliberately does not exist — only the severity and search-path difference vs. script.py matters
```

`script.py` (the trigger — PEP 723 header):

```python
#!/usr/bin/env -S uv run --quiet
# /// script
# requires-python = ">=3.11"
# dependencies = []
# ///
import sibling_package  # the same deliberately-nonexistent import, now in a PEP 723 script
```

```bash
mkdir ty-repro && cd ty-repro
# paste the three files above into this directory, then:

# A: baseline — bare auto-discovery, no environment.root override
uvx --from 'ty==0.0.72' ty check .
#   plain.py: warning[unresolved-import] (ty.toml's `warn` applied)
#   script.py: error[unresolved-import] (ty.toml ignored; no first-party search path)

# B: add `[environment]\nroot = ["."]` to ty.toml, SAME bare auto-discovery
uvx --from 'ty==0.0.72' ty check .
#   script.py: STILL error, STILL no first-party search path — the documented fix has NO effect

# C: identical ty.toml, but explicit (TY_CONFIG_FILE=ty.toml behaves the same)
uvx --from 'ty==0.0.72' ty check --config-file ty.toml .
#   script.py: warning[unresolved-import], first-party search path restored
```

The most probative detail in run A: `script.py`'s "Searched in the
following paths during module resolution" list is missing the
`/path/to/ty-repro (first-party code)` entry that `plain.py`'s list has — it
searches only the vendored stdlib and site-packages.

## Related issues (this defect is distinct from both)

- [#4083](https://github.com/astral-sh/ty/issues/4083) — 0.0.57 → 0.0.62:
  PEP-723-scoped diagnostics stopped honoring the project's `[rules]` severity
  overrides (confirmed intentional).
- [#4247](https://github.com/astral-sh/ty/issues/4247) — open proposal for a
  dedicated, warn-by-default local-import diagnostic for PEP 723 scripts.
- 0.0.71 → 0.0.72 additionally dropped the first-party search path (the
  containing directory) from PEP-723 scripts' module resolution —
  `environment.root` now defaults to empty, manufacturing brand-new
  `unresolved-import` diagnostics for first-party sibling imports that
  resolved fine right up to 0.0.71.

Both threads document `environment.root = ["."]` as the escape hatch, but
every example supplies it through a `pyproject.toml` `[tool.ty.*]` table.
This issue is specifically that the same hatch is inert when it comes from an
auto-discovered bare `ty.toml` — neither thread mentions that auto-discovery
doesn't count.

## Ask

Either auto-discovered config should apply `[environment] root` (and
`[rules]`) to PEP-723 scripts exactly as `--config-file`/`TY_CONFIG_FILE`
already does, or the documentation for the PEP-723 isolation escape hatch
should state explicitly that it takes effect only through an explicit
`--config-file`/`TY_CONFIG_FILE` and not through auto-discovery.

**Comment by MichaReiser** [comment]
Hi. Thanks for opening this issue. I'm currently working on PEP 723 script support and your feedback will help us shape the feature. 

I think the documentation can be improved here. What it suggests is that you add a `tool.ty.environment` section to your script, like so:

```python
#!/usr/bin/env -S uv run --quiet
# /// script
# requires-python = ">=3.11"
# dependencies = []
# 
# [tool.ty.environment]
# root = ["."]
# 
# [tool.ty.rules]
# unresolved-import = "warn"
# ///
import sibling_package  # the same deliberately-nonexistent import, now in a PEP 723 script
```

Our thinking here is that each script should work in isolation. They don't inherit any settings from the project. The thinking here is that you can share a script they should get the same experience as you (given they use the same uv/ty versions). 

But this is certainly an area where we're interested to get feedback on. Is repeating the configuration too annoying? Are there specific settings that must be changed often? Or are there specific settings that you would expect to be inherited? Should the ty settings be inherited if there's no `tool.ty` section? 

I can't promise that we'd add inheritance. An alternative is to allow an explicit `extend` or any other setting that opts-in to inheriting from the project. Or we might even decide to leave it as is.
